stanza/analysis/fitting.py

```
@@ -233,24 +238,36 @@ def compute_indices_from_threshold(
         fitted_current.max() - fitted_current.min()
     )
 
-    is_inverted = fitted_current[-1] < fitted_current[0]
-    return (
-        int(
-            np.argmax(
-                fitted_current <= min_threshold
-                if is_inverted
-                else fitted_current >= min_threshold
-            )
-        ),
-        int(
-            np.argmax(
-                fitted_current <= max_threshold
-                if is_inverted
-                else fitted_current >= max_threshold
-            )
-        ),
+    # Use magnitude to determine inversion when original current is negative
```

I would argue that if the current is inverted, then it is wrong.  Instead of providing a safe-guard for it, we should detect and fail early.